From 80df8f959af84ea3223ae9136ac4568c59420311 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sat, 3 Feb 2018 11:33:49 +0100
Subject: Fix include paths for secondary x86

This helps python find libraries and dependencies properly (openssl,
etc), so a few more modules are enabled.

It also fixes a crash because python would try to load the gcc2 libroot
even when built with gcc5.

diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
index d537963..fdbb8f4 100644
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -264,6 +264,7 @@ elif os.name == "posix":
                     directory = directory.replace('%A',
                         os.path.dirname(os.path.abspath(sys.argv[0] or os.getcwd())))
 
+                directory = os.path.join(directory, "x86")
                 if not os.path.isdir(directory):
                     continue
 
diff --git a/setup.py b/setup.py
index 21b0de2..33ba3f3 100644
--- a/setup.py
+++ b/setup.py
@@ -568,8 +568,8 @@ class PyBuildExt(build_ext):
         # Haiku-specific library locations
         if host_platform == 'haiku1':
             inc_dirs += ['/boot/develop/headers/posix',
-                         '/boot/system/develop/headers']
-            lib_dirs += ['/boot/system/develop/lib']
+                         '/boot/system/develop/headers/x86']
+            lib_dirs += ['/boot/system/develop/lib/x86']
 
         # OSF/1 and Unixware have some stuff in /usr/ccs/lib (like -ldb)
         if host_platform in ['osf1', 'unixware7', 'openunix8']:
-- 
2.15.1

